src/main/java/com/google/sps/servlets/StreamServlet.java

```diff
@@ -31,17 +30,34 @@
 @WebServlet("/api/v1/stream")
 public class StreamServlet extends HttpServlet {
 
+  DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
     Stream stream = new Stream();
     String jsonStream = convertToJson(stream);
     response.setContentType("application/json");
-    response.getWriter().println(jsonStream);  
+    response.getWriter().println(jsonStream); 
   }
 
   private String convertToJson(Stream stream) {
     Gson gson = new Gson();
     String json = gson.toJson(stream);
     return json;
   }
+
+  public ArrayList<Entity> appReader(ArrayList<String> appIds) {
+    ArrayList<Entity> apps = new ArrayList<Entity>();
+    Query query = new Query("App");
+    PreparedQuery results = datastore.prepare(query);
+
+    for(String id: appIds) {
+        for(Entity entity: results.asIterable()){
+            if(((String)entity.getProperty("id")).equals(id)){
+                apps.add(entity);
+            }
+        }
+    }
+    return apps;
+  }
```

Comment:
This should be in its own "AppReader" class. Also add tests

Reply:
+1

maybe it should go in google/sps/service